lib/system_message.rb

```diff
@@ -43,7 +43,7 @@ def create(type, params = {})
 
     post = I18n.with_locale(@recipient.effective_locale) { creator.create }
 
-    DiscourseEvent.trigger(:system_message_sent, post: post, message_type: type)
+    DiscourseEvent.trigger(:system_message_sent, post: post, message_type: type, target_user: @recipient)
```

Comment:
Are we able to get the target_user from the post object instead? The more "custom" attributes we have to support in an event trigger, the harder it is to change it in the future.